### PR TITLE
Add syntax highlight support for zeek and sig

### PR DESCRIPTION
**Pull request by AmazingPP**
Better syntax highlighting than  [bro/zeek - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=aeppert.bro)

QQ20220720-204210@2x

> [syntax_test.zeek.zip](https://github.com/bbannier/zeek-language-server/files/9149775/syntax_test.zeek.zip)

### CODE REVIEW

**Comment by bbannier** [comment]
This is great, but like I mentioned offline probably out of scope for this extension. I think this would fit much better as a patch for https://github.com/aeppert/bro-vsc-extension which seems to be _the syntax highlighting extensions for Zeek_ for vscode. Could you please try to get a PR into that extension first? If the author is unresponsive we could revisit pulling vscode only syntax it into this extension. In any case, we could add a dependencies to it to make this extension more useful out of the box (i.e., add `﻿"extensionDependencies": ["aeppert.bro"]` to `package.json`).

If we'd implement it here I would much rather see us implement support for [LSP semantic tokens](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) (probably support for the `full` flavor might be good enough).

**Comment by AmazingPP** [comment]
In fact, I created RP ([New features](https://github.com/aeppert/bro-vsc-extension/pull/4)) four months ago for [bro-vsc-extension](https://github.com/aeppert/bro-vsc-extension), but no one replied to me.

I think syntax highlight and semantic highlight can exist at the same time. When LSP is not working, syntax highlight can provide basic highlight. Just like this extension [F# Language Server](https://github.com/georgewfraser/fsharp-language-server)

When LSP works normally, semantic highlight can provide better highlight than syntax highlight. because The editor applies the highlighting from semantic tokens on top of the highlighting from grammars.

**Review by bbannier — CHANGES_REQUESTED** [review]
> In fact, I created RP (https://github.com/aeppert/bro-vsc-extension/pull/4) four months ago for [bro-vsc-extension](https://github.com/aeppert/bro-vsc-extension), but no one replied to me.

Fair enough, let's then merge this into this one.

I left a couple of mostly minor comments; would be great if you could have a look.

The pre-commit hook CI also flagged a couple of formatting issues which should be easy to fix by either copying the feedback from CI or by running `pre-commit` locally (you can disable checks which need Rust by running `SKIP=fmt,clippy,cargo-check pre-commit run  -a`).

File: vscode/syntaxes/zeek.json [hunk 1]
```diff
@@ -0,0 +1,419 @@
+{
```
Comment (bbannier) [inline vscode/syntaxes/zeek.json:1]:
```suggestion
{
    "$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
```

File: vscode/syntaxes/sig.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
+{
+    "name": "Signature",
+    "scopeName": "source.sig",
+    "fileTypes": [
+        "sig"
+    ],
+    "uuid": "d8db6740-ccd9-4e4a-9da3-8a3cf2313389",
```
[the remark this answers is not in the document]
Comment (AmazingPP) [reply vscode/syntaxes/sig.json:8]: The UUIDs were mainly used by TextMate to classify which file was a part of which bundle.

https://macromates.com/textmate/manual/bundles

File: vscode/syntaxes/zeek.json [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    "name": "Zeek",
+    "scopeName": "source.zeek",
+    "fileTypes": [
+        "bro",
+        "zeek"
+    ],
+    "uuid": "d27211b1-e07c-4c8d-9d7e-13daa805693c",
```
[the remark this answers is not in the document]
Comment (AmazingPP) [reply vscode/syntaxes/zeek.json]: https://github.com/bbannier/zeek-language-server/pull/11#discussion_r926702994

File: vscode/syntaxes/sig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bbannier) [reply vscode/syntaxes/sig.json]: I'd suggest to drop them until we figure out a need.

**Pull request merged**